dlc-manager/src/sub_channel_manager.rs

```diff
@@ -1079,50 +1110,139 @@ where
+                        };
+                        self.ln_chain_monitor.update_channel_funding_txo(
+                            &funding_txo,
+                            &c.ln_rollback.funding_outpoint,
+                            c.ln_rollback.channel_value_satoshis,
+                        )?;
+                        let commitment_transactions = Some(
+                            self.ln_chain_monitor
+                                .get_latest_holder_commitment_txn(&funding_txo)?,
+                        );
+                        let mut signed_subchannel = sub_channel.clone();
+                        signed_subchannel.state =
+                            SubChannelState::Signed(c.signed_subchannel.clone());
+                        self.force_close_signed_channel(
+                            signed_subchannel,
+                            commitment_transactions,
+                        )?;
+                        return Ok(());
+                    }
+                    SubChannelState::CloseOffered(ref c) => {
+                        let split_input = &c.signed_subchannel.split_tx.transaction.input[0];
+                        let funding_txo = lightning::chain::transaction::OutPoint {
+                            txid: split_input.previous_output.txid,
+                            index: split_input.previous_output.vout as u16,
+                        };
+                        let commitment_transactions = Some(
+                            self.ln_chain_monitor
+                                .get_latest_holder_commitment_txn(&funding_txo)?,
+                        );
+                        let mut signed_subchannel = sub_channel.clone();
+                        signed_subchannel.state =
+                            SubChannelState::Signed(c.signed_subchannel.clone());
+                        self.force_close_signed_channel(
+                            signed_subchannel,
+                            commitment_transactions,
+                        )?;
+                        return Ok(());
+                    }
+                    SubChannelState::Signed(ref s) => {
+                        let split_input = &s.split_tx.transaction.input[0];
+                        let funding_txo = lightning::chain::transaction::OutPoint {
+                            txid: split_input.previous_output.txid,
+                            index: split_input.previous_output.vout as u16,
+                        };
+                        let commitment_transactions = Some(
+                            self.ln_chain_monitor
+                                .get_latest_holder_commitment_txn(&funding_txo)?,
+                        );
+                        self.force_close_signed_channel(sub_channel, commitment_transactions)?;
+                        return Ok(());
+                    }
+                    _ => return Ok(()),
+                }
             }
+            _ => return Ok(()),
```

Comment:
I think it would be good to add the variants explicitly so that we don't miss handling important states if new variants are added as the code changes (including LDK!).

Reply:
I'm not convinced it's very useful but it doesn't hurt so added.

Reply:
It could be particularly useful in the case of a change in LDK (any dependency in general).

For instance, if we upgrade to 0.0.117 and they have added a new `ClosureReason` that we need to handle properly, the wildcard pattern would silently handle the new variant. Conversely, without the wildcard pattern the compiler would at least give us an error here and we would have to explicitly write code to handle the new variant.